Donor/Controllers/DonorController.cs

```diff
@@ -95,6 +95,51 @@ public async Task<IActionResult> GetAll()
             return Ok(donorDtos);
         }
 
+
+        /// <summary>
+        /// Update Edit
+        /// </summary>
+        /// <returns></returns>
+        [HttpPut("Update/{id}")]
+        public async Task<IActionResult> Update(int id, [FromBody] DonorDto donorDto)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            var existingDonor = await _repo.GetDonorByIdAsync(id);
```

Comment:
Suggestion :
Perhaps it's convenience to have a specific method to check this, something like this : 
public async Task<bool> ExistsAsync(int id)
{
	return await _context.Donor.AnyAsync(x => x.Id == id);
}

It's up to you!